bot/bot.py

```diff
@@ -39,6 +45,69 @@ def __init__(self, redis_session: RedisSession, **kwargs):
         self.loop.create_task(self.check_channels())
         self.loop.create_task(self.send_log(self.name, "Connected!"))
 
+        # Mapping for the main games leaderboard
+        # The key holds the name of the discord Cog and the value is the redis cache object
+        # for that specific game leaderboard
+        self.games_leaderboard: dict[str, tuple[RedisCache, ...]] = dict()
+        self.auto_per_day_leaderboard_clear.start()
+
+    @loop()
+    async def auto_per_day_leaderboard_clear(self) -> None:
+        """Loop for clearing the per day leaderboard redis cache at UTC midnight."""
+        # once d.py get support for `time` parameter in loop decorator,
+        # this can be removed and the loop can use the `time=datetime.time.min` parameter
+        now = datetime.utcnow()
+        tomorrow = now + timedelta(days=1)
+        midnight_tomorrow = tomorrow.replace(hour=0, minute=0, second=0)
+
+        await sleep_until(midnight_tomorrow)
+
+        for _, per_day_lb in self.games_leaderboard.values():
+            await per_day_lb.clear()
+
+    def ensure_leaderboard(self, cog: Cog) -> None:
+        """Ensure that per day leaderboard and total leaderboard redis cache are available for the cog."""
+        if self.games_leaderboard.get(cog.qualified_name):
+            return
+
+        # This cache contains the leaderboard for game `cog`
+        # RedisCache[int: int]
+        # Where the key is the discord user ID, and the values if the total
+        # points registered for that user.
+        _leaderboard_cache = RedisCache(namespace=f"{cog.qualified_name}-total")
+
+        # This cache contains the per day leaderboard, same specifies
+        _per_day_leaderboard = RedisCache(namespace=cog.qualified_name)
+
+        self.games_leaderboard[cog.qualified_name] = (_leaderboard_cache, _per_day_leaderboard)
+
+    async def increment_points(self, cog: Cog, user: discord.Member, points: int = DEFAULT_POINTS) -> None:
+        """
+        Increment points for a user in the leaderboard for game `cog` by `points`.
+
+        If max points for today are hit for that particular game cog, then set today's points to the max
+        points and add the difference of the(before today points and max possible points for a day to the
```

Comment:
I don't understand what you meant to do here

Reply:
So basically, a member can only get a maximum of `MAX_PER_GAME_PER_DAY_POINTS` points in a day, say I have `95` points now and I win a coinflip game, so I should get `105`points but since the max points are `100`, I can only earn `5` more points. Therefore I only get 5 more points, so in the total leaderboard, I get 5 more points which is basically `MAX_PER_GAME_PER_DAY_POINTS - current_points_today` as the docstring says.

Now, say I have `80` points, then I would get a total of `90` points and `current_total_points + points` would be the new total points as the day points are still less than the maximum.

Reply:
Yes I think I understand that but "and add the difference of the(before today points and max possible points for a day to the)" part. Even if you add a space that is not correct grammar.